File: backend/app/schemas/local_scan.py
from pydantic import BaseModel


class LocalFileSummary(BaseModel):
    id: int
    filename: str
    path: str
    size_bytes: int
    matched_title: str | None = None
    matched_author: str | None = None
    matched_score: float | None = None


class FileIdsRequest(BaseModel):
    file_ids: list[int]


class CopyResult(BaseModel):
    copied: int
    failed: int


class DismissResult(BaseModel):
    dismissed: int

File: frontend/src/types/localScan.ts
export interface LocalFileSummary {
  id: number
  filename: string
  path: string
  size_bytes: number
  matched_title: string | null
  matched_author: string | null
  matched_score: number | null
}

export interface CopyResult {
  copied: number
  failed: number
}

export interface DismissResult {
  dismissed: number
}
